### PR TITLE
fix: calculate prometheus disks_offline/disks_total correctly

### LINKED ISSUE #11196: MinIO disks_offline and disks_total in Prometheus monitoring
Hello, I have minio cluster with 5 nodes and 2 disks for each. I try to monitor it with the help of /minio/prometheus/metrics.  

## Expected Behavior
if one or more disk is unreachable in `mc info` I want to see it in prometheus\grafana\alertmanager

## Current Behavior
prom_minio_monitoring

## Possible Solution
Just want to know if it is a bug. is it possible to see this info in /minio/prometheus/metrics?

## Steps to Reproduce (for bugs)
```
Minio start command: /usr/local/bin/minio server https://minio-0{1...5}.domain.com/var/minio/data{1...2}
Environment variables: 
MINIO_ACCESS_KEY=abc
MINIO_SECRET_KEY=12345678
MINIO_PROMETHEUS_AUTH_TYPE="public"
```

## Your Environment

* Version used (`minio --version`): minio version RELEASE.2020-12-29T23-29-29Z
* Operating System and version (`uname -a`): NAME="Red Hat Enterprise Linux Server" VERSION="7.9 (Maipo)" Linux minio-01 3.10.0-1160.el7.x86_64 #1 SMP Tue Aug 18 14:50:17 EDT 2020 x86_64 x86_64 x86_64 GNU/Linux

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: calculate prometheus disks_offline/disks_total correctly

## Motivation and Context
fixes #11196

## How to test this PR?
As per #11196 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [x] Unit tests needed

### CODE REVIEW

**Review by poornas — APPROVED** [review]
LGTM

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#d21ee154-3fdd-4f29-b612-273ac8a85b81) |
<a name="d21ee154-3fdd-4f29-b612-273ac8a85b81"></a>
### 11215-532a3a2/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31167
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1eac1f44fdbc:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 10 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 38 seconds
(4/15) Running aws-sdk-ruby tests ... done in 21 seconds
(5/15) Running awscli tests ... done in 2 minutes and 56 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 3 minutes and 44 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 51 seconds
(9/15) Running minio-go tests ... done in 6 minutes and 22 seconds
(10/15) Running minio-java tests ... FAILED in 8 minutes and 57 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[user metadata]",
  "duration": 175,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = AuthenticationFailed, message = -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=AuthenticationFailed) =====\nDescription=Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\nRequestId:09678c71-701e-00ff-045a-e2d076000000\nTime:2021-01-04T05:31:49.1102374Z, Details: \n   AuthenticationErrorDetail: The MAC signature found in the HTTP request 'PG8XplSK9Mrmzpw8ovaibxMoU6c4PnncxbxNPFg/1uo=' is not the same as any computed signature. Server used following string to sign: 'PUT\n\n\n128\n\napplication/xml\n\n\n\n\n\n\nx-ms-blob-cache-control:\nx-ms-blob-content-disposition:\nx-ms-blob-content-encoding:\nx-ms-blob-content-language:\nx-ms-blob-content-type:application/octet-stream\nx-ms-client-request-id:c2aa7aa5-9e7b-4e94-4a4b-41fcec34239f\nx-ms-date:Mon, 04 Jan 2021 05:31:49 GMT\nx-ms-meta-my_header1:a   b   c\nx-ms-meta-my_header2:\"a   b   c\"\nx-ms-meta-my_project:Project One\nx-ms-meta-my_unicode_tag:å•†å“\nx-ms-version:2019-02-02\n/minioazure/minio-java-test-ih0m1r/minio-java-test-265d132\ncomp:blocklist\ntimeout:1501'.\n   Code: AuthenticationFailed\n   PUT https://minioazure.blob.core.windows.net/minio-java-test-ih0m1r/minio-java-test-265d132?comp=blocklist&timeout=1501\n   Authorization: REDACTED\n   Content-Length: [128]\n   Content-Type: [application/xml]\n   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2021-01-04T04:59:32Z]\n   X-Ms-Blob-Cache-Control: []\n   X-Ms-Blob-Content-Disposition: []\n   X-Ms-Blob-Content-Encoding: []\n   X-Ms-Blob-Content-Language: []\n   X-Ms-Blob-Content-Type: [application/octet-stream]\n   X-Ms-Client-Request-Id: [c2aa7aa5-9e7b-4e94-4a4b-41fcec34239f]\n   X-Ms-Date: [Mon, 04 Jan 2021 05:31:49 GMT]\n   X-Ms-Meta-My_header1: [a   b   c]\n   X-Ms-Meta-My_header2: [\"a   b   c\"]\n   X-Ms-Meta-My_project: [Project One]\n   X-Ms-Meta-My_unicode_tag: [商品]\n   X-Ms-Version: [2019-02-02]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 403 Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\n   Content-Length: [1091]\n   Content-Type: [application/xml]\n   Date: [Mon, 04 Jan 2021 05:31:48 GMT]\n   Server: [Microsoft-HTTPAPI/2.0]\n   X-Ms-Error-Code: [AuthenticationFailed]\n   X-Ms-Request-Id: [09678c71-701e-00ff-045a-e2d076000000]\n\n\n, bucketName = minio-java-test-ih0m1r, objectName = minio-java-test-265d132, resource = /minio-java-test-ih0m1r/minio-java-test-265d132, requestId = 1656F07699246F69, hostId = c2264c44-e79d-4d90-a7ce-a435da8abb3f)\nrequest={method=PUT, url=http://minio-c3.minio.io:31167/minio-java-test-ih0m1r/minio-java-test-265d132, headers=x-amz-meta-My-Unicode-Tag: 商品\nx-amz-meta-My-Project: Project One\nx-amz-meta-My-header1: a   b   c\nx-amz-meta-My-Header2: \"a   b   c\"\nContent-Type: application/octet-stream\nHost: minio-c3.minio.io:31167\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: A9oFTxee7YVcJ9fWsgQeKg==\nx-amz-content-sha256: 1ff7959f86334ddc5c188a5083268f600146328b2b6c5185e75bf7d9387d6b74\nx-amz-date: 20210104T053148Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210104/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date;x-amz-meta-my-header1;x-amz-meta-my-header2;x-amz-meta-my-project;x-amz-meta-my-unicode-tag, Signature=*REDACTED*\n}\nresponse={code=403, headers=Accept-Ranges: bytes\nContent-Length: 3082\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 1656F07699246F69\nX-Xss-Protection: 1; mode=block\nDate: Mon, 04 Jan 2021 05:31:49 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.putObject(MinioClient.java:4547), io.minio.MinioClient.putObject(MinioClient.java:2713), io.minio.MinioClient.putObject(MinioClient.java:2830), FunctionalTest.testPutObject(FunctionalTest.java:763), FunctionalTest.putObject(FunctionalTest.java:890), FunctionalTest.runObjectTests(FunctionalTest.java:3751), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(10/15) Running minio-js tests ... done in 2 minutes and 41 seconds
(11/15) Running minio-py tests ... done in 18 minutes and 23 seconds
(12/15) Running s3cmd tests ... done in 2 minutes and 16 seconds
(13/15) Running s3select tests ... done in 59 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/metrics.go [hunk 1]
```diff
@@ -497,10 +497,11 @@ func storageMetricsPrometheus(ch chan<- prometheus.Metric) {
...
 	}
 
-	// Fetch disk space info, ignore errors
-	storageInfo, _ := objLayer.StorageInfo(GlobalContext, true)
+	server := getLocalServerProperty(globalEndpoints, &http.Request{
+		Host: GetLocalPeer(globalEndpoints),
+	})
 
-	onlineDisks, offlineDisks := getOnlineOfflineDisksStats(storageInfo.Disks)
+	onlineDisks, offlineDisks := getOnlineOfflineDisksStats(server.Disks)
```
Comment (vadmeste) [inline cmd/metrics.go:504]: This is a breaking change, but I like that for prometheus since users generally want to have information per server

**Review by vadmeste — COMMENTED** [review]
One comment, but LGTM otherwise

File: cmd/metrics.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/metrics.go:504]: It's not breaking it's still the value represented per server.

**Comment by akong77** [comment]
Hello, I have the same problem.The prometheus always show offline.But mc admin info show no offline.How to fix it?

**Pull request merged**